cpp/src/arrow/flight/sql/client.h

```diff
@@ -450,6 +455,18 @@ class ARROW_FLIGHT_SQL_EXPORT Transaction {
   std::string transaction_id_;
 };
 
+/// \brief ADBC entry point.
+///
+/// A pointer to this function can be manually passed to the ADBC
+/// driver manager to initialize the ADBC driver.  For example, if you
+/// are already linking to Arrow, instead of asking the driver manager
+/// to reload Arrow to get the Flight SQL driver, you can just pass
```

Comment:
I'm not sure "reload Arrow" is a supported operation, given that we may have globals/statics with non-trivial constructors. Should we advertise that possibility at all?

Reply:
I rewrote the docstring to be shorter and not advertise that pitfall.